### PR TITLE
add resource_type validator and specs

### PR DESCRIPTION
**Pull request by csyversen**
fixes part of https://github.com/nulib/next-generation-repository/issues/483

i figured i'd go with smaller PRs with a feature in each one

### CODE REVIEW

File: app/models/resource_type_validator.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class ResourceTypeValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    invalid_resources = get_invalid_resources(value)
+    record.errors.add(attribute, "#{invalid_resources.join(', ')} is/are not a valid resource type(s)") if invalid_resources.any?
```
Comment (kdid) [inline app/models/resource_type_validator.rb]: use inflections here rather than "is/are" or "(s)"

File: app/models/resource_type_validator.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  private
+
+    def get_invalid_resources(value)
+      authority = Qa::Authorities::Local::FileBasedAuthority.new('resource_types')
+      # the .present? check is to accomodate empty string/nil values since this
+      # isn't a required field, so ""/nil are valid entries
+      invalid_entries = value.map { |x| x if authority.search(x).blank? && x.present? }
```
Comment (kdid) [inline app/models/resource_type_validator.rb]: I don't think this is doing what you want. This will validate part of a two-word value. For example, the word "Part" will validate. `Hyrax::ResourceTypesService.select_options` is already a thing you could just check against allowed values, also the `authority` you are using has some methods that would work too. I think just skipping if blank/nil and then check if included in the valid options would be alot clearer logic.

File: app/models/resource_type_validator.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  def validate_each(record, attribute, value)
+    invalid_resources = get_invalid_resources(value)
+    record.errors.add(attribute, "#{invalid_resources.join(', ')} is/are not a valid resource type(s)") if invalid_resources.any?
+  end
+
+  private
+
+    def get_invalid_resources(value)
+      authority = Qa::Authorities::Local::FileBasedAuthority.new('resource_types')
+      # the .present? check is to accomodate empty string/nil values since this
```
Comment (kdid) [inline app/models/resource_type_validator.rb]: remove comment and add spec to test a blank value is valid instead

File: spec/models/resource_type_validator_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ResourceTypeValidator do
```
Comment (kdid) [inline spec/models/resource_type_validator_spec.rb]: add spec to test blank/nil (see above)

**Comment by bmquinn** [comment]
@csyversen Also, can you move `resource_type_validator.rb` to `app/models/concerns`? That's where the other is, and I think it makes sense to keep them all together.

**Comment by kdid** [comment]
I approved. Just to clarify, for the inflector, I was thinking to use the Rails API and add it here instead of writing a helper method (unless this is not still a current rails good practice):

https://github.com/nulib/donut/blob/master/config/initializers/inflections.rb#L9. 

then you could just do something like: 
`'is'.pluralize(count)`

This would also be reusable if we write more validators in the upcoming days that use the same language...
